app/Http/Controllers/FenneluckyController.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Fennelucky;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class FenneluckyController extends Controller
+{
+    public function store(Request $request): JsonResponse
+    {
+        $request->validate([
+            'name' => 'required|unique:fenneluckies,name,NULL,id,user_id,'.auth()->id(),
```

Comment:
This rule makes my head spin a little 😆 Consider refactoring to a FormRequest and turning the rules into an array instead of `|`-separated.

Does `unique` even accept so many arguments?

Instead of `auth()->id()`, consider using `$request->user()->id`. This verifies you're only doing things with the user that sent the request (retrieved based on the request's token).

Reply:
I created a new request type for the validation to keep the controller clean.